### PR TITLE
Feat/typos and polish

### PR DESCRIPTION
**Pull request by aeschi**
Lot's of small changes in pairing with @HenrietteN 
[currently listed in PO review](https://app.asana.com/0/1204152181784036/1204153176071406)

### CODE REVIEW

File: src/components/buttons/primary-destructive-loading.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+			onClick={onClick}
+			type={type}
+		>
+			<div className={`-translate-x-2 ${isLoading ? "flex" : "hidden"}`}>
+				<SpinnerIcon />
+			</div>
+			<span className="flex flex-row items-center gap-3">{label}</span>
+		</button>
+	);
+};
```
Comment (raphael-arce) [inline src/components/buttons/primary-destructive-loading.tsx:33]: as already discussed, I think we can add the loading feature to our existing buttons 👍

File: src/components/tree-detail/last-waterings/watering-card.tsx [hunk 2]
```diff
@@ -25,12 +25,15 @@ export const WateringCard: React.FC<WateringCardProps> = ({ wateringData }) => {
...
 	const { deleteWatering } = useWaterTree();
+	const [isDeleteWateringLoading, setIsDeleteWateringLoading] = useState(false);
 
 	const [isConfirmDeleteVisible, setIsConfirmDeleteVisible] = useState(false);
 	const isWateringByUser = wateringData.username === username;
 
 	const onClickDelete = async () => {
+		setIsDeleteWateringLoading(true);
 		await deleteWatering(wateringData.id);
+		setIsDeleteWateringLoading(false);
```
Comment (raphael-arce) [inline src/components/tree-detail/last-waterings/watering-card.tsx:36]: I'd suggest to remove the `wateringLoading` state inside the `useWaterTree()` hook at the same time 👍

File: src/components/tree-detail/tree-water-needs/watering-dialog.tsx [hunk 3]
```diff
@@ -26,7 +27,10 @@ export const WateringDialog: React.FC = () => {
 			const date = new Date(e.currentTarget.date.value);
 
 			try {
+				setIsWateringLoading(true);
 				await waterTree(amount, date);
+				setIsLastWateringsExpanded(true);
```
Comment (raphael-arce) [inline src/components/tree-detail/tree-water-needs/watering-dialog.tsx:33]: should we expand the lastWaterings even if waterTree wasn't successful?

File: src/components/tree-detail/tree-water-needs/watering-dialog.tsx [hunk 4]
```diff
@@ -15,8 +15,9 @@ export const WateringDialog: React.FC = () => {
 	const i18n = useI18nStore().i18n();
 	const { waterTree } = useWaterTree();
 	const { handleError } = useErrorStore();
-
 	const formattedToday = format(new Date(), "yyyy-MM-dd");
+	const [isWateringLoading, setIsWateringLoading] = useState(false);
+	const { setIsLastWateringsExpanded } = useTreeStore();
```
Comment (raphael-arce) [inline src/components/tree-detail/tree-water-needs/watering-dialog.tsx:21]: looks like this is missing an import?

**Review by raphael-arce — APPROVED** [review]
I have a few suggestions otherwise lgtm 👍

File: src/components/tree-detail/tree-water-needs/watering-dialog.tsx (hunk 3 above)
Comment (aeschi) [reply src/components/tree-detail/tree-water-needs/watering-dialog.tsx:33]: I guess it could make sense to show that "nothing" was added

**Pull request merged**